lib/src/iteration/iteration.dart

```diff
@@ -31,6 +31,14 @@ Iterable<DateTime> getRecurrenceRuleInstances(
   var count = rrule.count;
 
   var currentStart = start;
+
+  if (count == null && after != null) {
+    /// Shortcut for not calculating unnecessary recurrences.
+    final skippableDifference = after.difference(start);
+
+    currentStart = currentStart.add(skippableDifference);
```

Comment:
```suggestion
    // Shortcut for not calculating unnecessary recurrences.
    currentStart = after;
```

Reply:
Ah, you are right, guess I was overthinking it. :smile: